source/request-as-event-emitter.js

```diff
@@ -74,7 +74,7 @@ module.exports = options => {
 				redirectUrl = (new URLGlobal(bufferString, urlLib.format(options))).toString();
```

Comment:
I think using `urlLib.format` to get the baseUrl has the same problem as was discussed in https://github.com/sindresorhus/got/pull/519#issuecomment-404950951.

I can take a look at adding a test and fixing today.

Reply:
The baseUrl won’t have the right query string, but I don’t think there’s any value for new url for which it would affect the result. So this seems fine.